contracts/KyberPayWrapper.sol

```diff
@@ -0,0 +1,181 @@
+pragma solidity 0.4.18;
+
+
+import "./ERC20Interface.sol";
+import "./Withdrawable.sol";
+import "./ReentrancyGuard.sol";
+
+
+interface KyberNetwork {
+    function tradeWithHint(
+        ERC20 src,
+        uint srcAmount,
+        ERC20 dest,
+        address destAddress,
+        uint maxDestAmount,
+        uint minConversionRate,
+        address walletId,
+        bytes hint)
+    external
+    payable
+    returns(uint);
+}
+
+
+contract KyberPayWrapper is Withdrawable, ReentrancyGuard {
+    ERC20 constant public ETH_TOKEN_ADDRESS = ERC20(0x00eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee);
+
+    struct PayData {
+        ERC20 src;
+        uint srcAmount;
+        ERC20 dest;
+        address destAddress;
+        uint maxDestAmount;
+        uint minConversionRate;
+        address walletId;
+        bytes paymentData;
+        bytes hint;
+        KyberNetwork kyberNetworkProxy;
+    }
+
+    function () public payable {}
+
+    event ProofOfPayment(address _beneficiary, address _token, uint _amount, bytes _data);
+
+    function pay(
+        ERC20 src,
+        uint srcAmount,
+        ERC20 dest,
+        address destAddress,
+        uint maxDestAmount,
+        uint minConversionRate,
+        address walletId,
+        bytes paymentData,
+        bytes hint,
+        KyberNetwork kyberNetworkProxy
+    ) nonReentrant public payable
+    {
+
+        require(src != address(0));
+        require(dest != address(0));
+        require(destAddress != address(0));
+
+        if (src == ETH_TOKEN_ADDRESS) require(srcAmount == msg.value);
+
+        PayData memory payData = PayData(
+            src,
+            srcAmount,
+            dest,
+            destAddress,
+            maxDestAmount,
+            minConversionRate,
+            walletId,
+            paymentData,
+            hint,
+            kyberNetworkProxy
+        );
+
+        uint paidAmount = (src == dest) ? doPayWithoutKyber(payData) : doPayWithKyber(payData);
+
+        // log as event
+        ProofOfPayment(destAddress, dest, paidAmount, paymentData);
```

Comment:
`beneficiary` is actually `msg.sender`.

Reply:
as discussed will now have payer and payee in the event.